src/server/middleware/auth/gmeauth.js

```diff
@@ -167,6 +166,72 @@ function GMEAuth(session, gmeConfig) {
             .nodeify(callback);
     }
 
+    function _prepareAdminAccount(callback) {
+        var admin = gmeConfig.authentication.admin,
+            pieces,
+            adminId,
+            password;
+
+        if (!admin) {
+            return Q();
+        }
+
+        if (!gmeConfig.authentication.enable) {
+            logger.warn('gmeConfig.authentication.admin is specified but auth is disabled - will not create account!');
+            return Q();
+        }
+
+        pieces = admin.split(':');
+        adminId = pieces[0];
+        password = pieces[1];
+
+        return collection.findOne({_id: adminId})
+            .then(function (userData) {
+                if (userData) {
+                    logger.debug('Admin user exists [' + adminId + ']');
+                } else {
+                    password = password || (chance.word({syllables: 4}) + chance.natural({min: 1, max: 999}));
+                    logger.warn('Creating admin "' + adminId + '" with password: ' + password);
+                    logger.warn('To change password login at the profile page or use ' +
+                        'webgme-engine/src/bin/usermanager.js');
+
+                    return addUser(adminId, admin, password, true, {
+                        overwrite: true,
+                        siteAdmin: true,
+                        guestOrAdmin: true
+                    });
+                }
+            })
+            .nodeify(callback);
+    }
+
+    function _preparePublicOrganizations(callback) {
+        var publicOrgs = gmeConfig.authentication.publicOrganizations;
+
+        if (publicOrgs.length === 0) {
+            return Q();
+        }
+
+        if (!gmeConfig.authentication.enable) {
+            logger.warn('gmeConfig.authentication.publicOrgs is specified but auth is disabled - ' +
+                'will not create organizations');
+            return Q();
+        }
+
+        return Q.all(publicOrgs.map(function (orgId) {
+            return collection.findOne({_id: orgId})
+                .then(function (userData) {
+                    if (userData) {
+                        logger.debug('Public organization exists [' + orgId + ']');
+                    } else {
+                        logger.info('Creating public organization "' + orgId + '"');
```

Comment:
I know that this is not a warning conceptually, but shouldn't this be on the same level of logging as the admin user creation?